### PR TITLE
Disable 'verify if module is latest' for shops under 1.7.0.0 for 4.14.x and bump version to 4.14.3

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Disable 'verify if module is latest' for shops under 1.7.0.0 for 4.14.x and bump version to 4.14.3
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      | See below

## Context

The upgrade module has a feature where it checks whether there is a more recent version of itself and if it does, it tells the user "please use the latest version, you are using an outdated version".

Capture d’écran 2023-09-08 à 23 09 45

Currently if someone installs the module 4.14.2 on a shop, the module will detect there is a 4.16.3 version and says to user "please don't use me! use the latest 4.16.3 version instead!"

Consequently this PR disables this check, this was a suggestion from @Hlavtox.

## How to test

Create a test autoupgrade release from this PR. Install this test module
- on a shop 1.6.1.24 : you do NOT see the warning `Your current version of the module is out of date. Update now` (because this PR disables the check on versions under 1.7.0.0)
- on a shop 1.7.8.10 : you see the warning `Your current version of the module is out of date. Update now`

> [!IMPORTANT]  
> Did you notice target branch is `4.14.x`? So if you rebase the PR against target branch with `git prc` be careful to select the right branch

### CODE REVIEW

**Comment by matks** [comment]
https://github.com/PrestaShop/autoupgrade/pull/626/commits/bd9fc65f5b78dc97db4388de087e03f45c36c182
As the branch `4.14.x` is only for updates 1.6 to 1.7 I set CI to test againt latest 1.7.x version

**Comment by Hlavtox** [comment]
@matks It should not perform this check for versions lower than 1.7, not lower than 8.

For people on 1.7, we want them to upgrade to 4.16. :)

**Comment by matks** [comment]
True. I guess sending this PR at 22pm on a Friday evening was not the best way to obtain a good result. I will submit the changes.

**Comment by matks** [comment]
@Hlavtox Fixed in https://github.com/PrestaShop/autoupgrade/pull/626/commits/1a92a6b020a15d50d6a677edca4585f27ff16373

**Review by kpodemski — APPROVED** [review]

**Comment by matks** [comment]
For your information, I discussed with @Robin-Fischer-PS and @MatShir and this is what we suggest for release 4.14.3:

- this PR test is OK, no need for further release testing as the only difference between 4.14.2 and 4.14.3 will be this PR
- upgrading from 1.6.x to 1.7.x shops is not supported anymore so this 4.14.3 release will be out of scope
- this means that 4.14.3 will be published and available for anyone to use but we make no warranties, express or implied
- basically the release will be provided "as is", without warranty

**Review by aniszr — APPROVED** [review]
Hello @matks 

I tested your PR with two PS versions as mentioned :

- **PS 1.6.1.24**
The checks on versions of Autoupgrade module are **disabled** :heavy_check_mark: 

See attached screenshot:

image

image


- **PS 1.7.8.10**
The checks on versions of Autoupgrade module are **enabled** :heavy_check_mark: 

See attached screenshot:
image

image


Qa :heavy_check_mark: 

Thank you!

**Comment by MatShir** [comment]
Hi @aniszr, 
The check is inside the module (not in the module list). When you want to upgrade the shop, the module check if there is a new version available.

**Comment by aniszr** [comment]
Hello @MatShir 

Thanks for your feedback !

See attachet screenshot for **PS 1.7.8.10**

image

Thank you!

**Comment by MatShir** [comment]
👌 sweat ! thanks @aniszr we can merge and release the version !
